apps/wing-console/console/ui/src/services/use-dynamodb-table.ts

```diff
@@ -0,0 +1,47 @@
+import { useCallback, useMemo } from "react";
+
+import { trpc } from "./trpc.js";
+
+export interface UseTableOptions {
+  resourcePath: string;
+}
+
+export const useDynamodbTable = ({ resourcePath }: UseTableOptions) => {
+  const table = trpc["dynamodb-table.info"].useQuery({ resourcePath });
+  const tableInsert = trpc["dynamodb-table.insert"].useMutation();
+  const tableDelete = trpc["dynamodb-table.delete"].useMutation();
+
+  const addRow = useCallback(
+    async (row: any) => {
+      await tableInsert.mutateAsync({
+        resourcePath,
+        data: row,
+      });
+    },
+    [tableInsert, resourcePath],
+  );
+
+  const removeRow = useCallback(
+    async (index: number) => {
+      if (!table.data?.rows[index]) {
+        return;
+      }
+      await tableDelete.mutateAsync({
+        resourcePath,
+        data: table.data.rows[index] || {},
+      });
```

Comment:
I'd recommend using a variable to store the row so you don't have to add `|| {}`.

 ```suggestion
      const row = table.data?.rows[index];
      if (!row) {
        return;
      }
      await tableDelete.mutateAsync({
        resourcePath,
        data: row,
      });
```